torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1577,11 +1577,34 @@ std::string ProcessGroupNCCL::getNCCLWatchdogDebugInfo() {
   return retrieveDesyncReport(store_, "NCCL", rank_, size_);
 }
 
+// We want to have both PG ID and PG name for the logging prefix.
+// The reason why we have this two is because PG ID records how many
+// ProcessGroupNCCL objects were created on a specific rank and this
+// is useful for folks to know how many NCCL-PGs created on a certain
+// rank and what parallelism which PG might belong to in debugging.
+// while PG name (or group name) is set in `setGroupName` by the passed in
+// `group_name` in `ProcessGroupNCCL::Options`.
+// The `group_name` is created in `_process_group_name` it will either be
+// a hash of all the ranks in the group or a counter of how many times
+// `_process_group_name` is called, essentially it means how many times we
+// have initialized PG in python layer. Before using split_group, even if
+// we are creating a new sub-PG, all ranks have to call the API at the same
+// time, and this makes `group_name` a unique identifier for a group (PG)
+// which makes it useful for debugging as well.
```

Comment:
For comm_split, yes, I verified in the unit test, pg_name and pg_id are kind same now. Because to create a sub-pg across ranks, there is only one python call. So in the comm_split case, it's not technically `global_uid` unless it is hashed. But I agree, the name is quite confusing here. And for PG0 if it is default pg, the hash will be same across ranks. (I guess you mean sub-PG right, if rank 0 and 1 are not in the same subPG, then yes, the hash will be different).

Reply:
> So in the comm_split case, it's not technically global_uid unless it is hashed.

is this intentional, or should we actually make it unique even for the comm split case?

Reply:
Good question, I don't know, maybe it's expected because now each rank only call the API once or fewer times, so there is no way for one to know how many more subPGs other ranks are going create.